tools/setup_discord.py

```
@@ -0,0 +1,413 @@
+    }
+
+
+def api(method: str, path: str, token: str, body: Optional[dict] = None) -> dict:
+    """Make a Discord API call, respecting rate limits automatically."""
+    url = f"{BASE}{path}"
+    data = json.dumps(body).encode() if body is not None else None
+    req = urllib.request.Request(url, data=data, headers=_headers(token), method=method)
+
+    for attempt in range(5):
+        try:
+            with urllib.request.urlopen(req) as resp:
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)